fix(rule): update async api rule (#86)
otto-de/api-guidelines: api-guidelines/async/contract/asyncapi/rules/must-provide-api-specification-using-asyncapi-for-asyncrhonous-apis.md

We use the [AsyncAPI](https://www.asyncapi.com/) specification as a standard to define contracts for asynchronous APIs.
Any event published for integration with other services must be described using the AsyncAPI specification.

The API description format used must be [AsyncAPI 3.0.0](https://www.asyncapi.com/docs/reference/specification/v3.0.0) or newer.

> We use extensions to extend the specification to our needs by using [specification extensions](https://www.asyncapi.com/docs/reference/specification/v3.0.0#specificationExtensions) in order to describe the functionality that we require.

Reviewer: The extension part seems duplicated :)

```suggestion
We extend the specification to our needs by using [specification extensions](https://www.asyncapi.com/docs/reference/specification/v3.0.0#specificationExtensions) in order to describe the functionality that we require.
```

> We publish the specification in [backstage](https://backstage.live.si.cloud.otto.de/startpage)

Reviewer: As external users might not be aware of the term "Backstage", I'd explain it. Furthermore, I was wondering if this is actually a MUST and if it is a MUST, shouldn't it rather be a dedicated rule? Currently it would be quite hard to find.

```suggestion
The specifications can be published in [our internal developer portal](https://backstage.live.si.cloud.otto.de/startpage)
```

Author: thanks for the feedback, I did not see the messages because of holiday, I will discuss quickly in the API COP about the MUST and if it should be a separate rule and otherwise do the changes

Author: Hey @cgebken we discussed in the API COP and the opinion was to keep the storage location inside the rule and also to have it as a MUST rule. I changed the naming though and so it should be ready for merging

Reviewer: I see. Thanks for the information. In that case I only wish for a period at the end of the sentence (see my latest comment) ;)

> We publish the specification in [our internal developer portal](https://backstage.live.si.cloud.otto.de/startpage)

Reviewer: ```suggestion
We publish the specification in [our internal developer portal](https://backstage.live.si.cloud.otto.de/startpage).
```
